go.mod

```diff
@@ -13,7 +13,7 @@ require (
 	github.com/google/go-cmp v0.5.8
 	github.com/google/uuid v1.1.2
 	github.com/hashicorp/go-azure-helpers v0.44.0
-	github.com/hashicorp/go-azure-sdk v0.20221014.1215546
+	github.com/hashicorp/go-azure-sdk v0.20221017.1134117
```

Comment:
just an FYI since `go-azure-sdk` tends to get updated pretty frequently, we tend to pull the SDK upgrade out into it's own PR (for example I've done this in #18833, but hadn't noticed this PR) - perhaps it'd be worth pulling the SDK upgrade out of this/rebasing this on top of #18833, so that we can merge that and focus on the SDK-switch here rather than the SDK upgrade?

Reply:
Sure! I did it here as I suspected one of the HC maintainers would pull in a new version quit soon and I can easily rebase my PR on it after that being merged.

Changes within the `go-azure-SDK` are coming in so often, hard to keep that in sync:)